src/lib/action/entry-transform-to-type.ts

```diff
@@ -0,0 +1,123 @@
+import EntryTransformToType from '../interfaces/entry-transform-to-type'
+import { APIAction } from './action'
+import { OfflineAPI } from '../offline-api'
+import Entry from '../entities/entry'
+import * as _ from 'lodash'
+
+class EntryTransformToTypeAction extends APIAction {
+  private fromFields?: string[]
+  private sourceContentTypeId: string
+  private targetContentTypeId: string
+  private transformEntryForLocale: (inputFields: any, locale: string) => Promise<any>
+  private identityKey: (fromFields: any) => Promise<string>
+  private shouldPublish: boolean|'preserve'
+  private removeOldEntries: boolean
+  private updateReferences: boolean
+
+  constructor (entryTransformation: EntryTransformToType) {
+    super()
+    this.fromFields = entryTransformation.from
+    this.sourceContentTypeId = entryTransformation.sourceContentType
+    this.targetContentTypeId = entryTransformation.targetContentType
+    this.identityKey = entryTransformation.identityKey
+    this.shouldPublish = entryTransformation.shouldPublish || false
+    this.removeOldEntries = entryTransformation.removeOldEntries || false
+    this.updateReferences = entryTransformation.updateReferences || false
+    this.transformEntryForLocale = entryTransformation.transformEntryForLocale
+  }
+
+  async applyTo (api: OfflineAPI) {
+    const entries: Entry[] = await api.getEntriesForContentType(this.sourceContentTypeId)
+    const locales: string[] = await api.getLocalesForSpace()
+
+    for (const entry of entries) {
+      const inputs = this.fromFields ? _.pick(entry.fields, this.fromFields) : entry.fields
+      const newEntryId = await this.identityKey(inputs)
+      const hasEntry = await api.hasEntry(newEntryId)
```

Comment:
This is a "feature" we borrowed from the regular entry-derive. It gives you the possibility to replace several entries to a single one or to be able to resume a migration if it has failed previously.

But I agree that this might also lead to unexpected behavior, so we can remove it and reintroduce again if necessary as a separate option.